src/org/thoughtcrime/securesms/jobs/MultiDeviceContactUpdateJob.java

```diff
@@ -245,7 +245,7 @@ private void generateFullContactUpdate()
   }
 
   private List<ContactData> getAllContacts() {
-    List<Address> contactAddresses = DatabaseFactory.getRecipientDatabase(context).getRegistered();
+    List<Address> contactAddresses = new ArrayList<>(DatabaseFactory.getRecipientDatabase(context).getAllAddresses());
```

Comment:
Does `getAllAddresses` filter out contacts we're not friends with and groups?

Reply:
No, but we filter out the groups in the loop after that with `if (!address.isPhone()) { continue; }`. And we filter out those non-friend contacts in `generateFullContactUpdate()`

Reply:
`getRegistered` just returns with an empty array, so I change to `getAllAddresses`...

Reply:
👍